scripts/trim-low-abund.py

```diff
@@ -175,122 +364,85 @@ def main():
         ct = khmer_args.create_countgraph(args)
 
     K = ct.ksize()
-    CUTOFF = args.cutoff
-    NORMALIZE_LIMIT = args.normalize_to
-
     tempdir = tempfile.mkdtemp('khmer', 'tmp', args.tempdir)
     print('created temporary directory %s; '
           'use -T to change location' % tempdir, file=sys.stderr)
 
+    trimmer = Trimmer(ct, not args.variable_coverage, args.cutoff,
+                      args.trim_at_coverage)
+    if args.diginorm:
+        trimmer.do_normalize = True
+        trimmer.diginorm_coverage = int(args.diginorm_coverage)
+
     # ### FIRST PASS ###
 
     save_pass2_total = 0
 
-    n_bp = 0
-    n_reads = 0
     written_bp = 0
     written_reads = 0
-    trimmed_reads = 0
 
     pass2list = []
     for filename in args.input_filenames:
+        # figure out temporary filename for 2nd pass
         pass2filename = os.path.basename(filename) + '.pass2'
         pass2filename = os.path.join(tempdir, pass2filename)
-        if args.output is None:
-            trimfp = get_file_writer(open(os.path.basename(filename) +
-                                          '.abundtrim', 'wb'),
-                                     args.gzip, args.bzip)
-        else:
-            trimfp = get_file_writer(args.output, args.gzip, args.bzip)
+        pass2fp = open(pass2filename, 'w')
 
-        pass2list.append((filename, pass2filename, trimfp))
+        # construct output filenames
+        outfp = args.output
+        if outfp is None:
+            outfp = open(os.path.basename(filename) + '.abundtrim', 'wb')
```

Comment:
I don't know how to use context managers here if 'outfp' is only sometimes created in the loop?